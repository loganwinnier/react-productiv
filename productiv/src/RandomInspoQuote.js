import { useState } from "react";
import { getInspoQuote } from "./inspoQuoteApi";
import Quote from './Quote';

/** Shows Button to generate new quote or Quote and button
 * State
 * -clicked: boolean to display initial button or not
 * -quote: object like {text: "string", author: "string"}
 * 
 * Header -> RandomInspoQuote -> Quote
 */

function RandomInspoQuote() {
  const [clicked, setClicked] = useState(false);

  /** gets new inspirational quote updates quote state */
  const newQuote = async () => {
    const quote = await getInspoQuote();
    setQuote(quote);
  };

  const [quote, setQuote] = useState(newQuote);

  /** Toggles display from initial button state */
  function handleGetQuote() {
    setClicked(true);
  }

  return (
    <div className="RandomInspoQuote">
      {
        clicked
          ?
          <>
            <Quote quote={quote} />
            <button onClick={newQuote}>Nü quøte!</button>
          </>
          :
          <button onClick={handleGetQuote}>Click here for an inspirational quøte!</button>
      }
    </div>
  );
}

export default RandomInspoQuote;

