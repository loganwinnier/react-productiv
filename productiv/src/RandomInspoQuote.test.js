import React from "react";
import { fireEvent, render } from "@testing-library/react";
import RandomInspoQuote from "./RandomInspoQuote";
// import { getInspoQuote } from "./inspoQuoteApi";


describe("RandomInspoQuote", function () {
    it("renders without crashing", function () {
        render(<RandomInspoQuote />);
    });

    it("contains Click here for an inspirational quøte button", function () {
        const result = render(<RandomInspoQuote />);
        expect(result.container.querySelector(".Quote")).toBeFalsy();
        expect(result.queryByText("Click here for an inspirational quøte!")).toBeInTheDocument();
    });

    it("Generates quote on inspirational quøte! button click", function () {
        // let getInspoQuote = jest.fn();
        // getInspoQuote.mockReturnValueOnce({ quote: { text: "I am a quote", author: "test" } });
        const result = render(<RandomInspoQuote />);
        expect(result.container.querySelector(".Quote")).toBeFalsy();

        const quoteButton = result.queryByText('Click here for an inspirational quøte!');
        fireEvent.click(quoteButton);
        expect(result.queryByText("Click here for an inspirational quøte!")).not.toBeInTheDocument();
        expect(result.container.querySelector(".Quote")).toBeTruthy();
        // expect(result.queryByText("I am a quote - test")).toBeInTheDocument();
        expect(result.queryByText("Nü quøte!")).toBeInTheDocument();

    });

    it("matches snapshot", function () {
        const { container } = render(<RandomInspoQuote />);
        expect(container).toMatchSnapshot();
    });
});