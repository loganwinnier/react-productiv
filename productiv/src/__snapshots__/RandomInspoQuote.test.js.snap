// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomInspoQuote matches snapshot 1`] = `
<div>
  <div
    class="RandomInspoQuote"
  >
    <button>
      Click here for an inspirational qu√∏te!
    </button>
  </div>
</div>
`;
